atlassian/bitbucket/cloud/base.py

```diff
@@ -7,10 +7,11 @@
 
 
 class BitbucketCloudBase(BitbucketBase):
-    def __init__(self, url, *args, **kwargs):
+    def __init__(self, url, link="self", *args, **kwargs):
```

Comment:
I'm wondering about this. Why is this needed?

Reply:
This is needed because the participants have no `link` attribute.
```
"participants": [
        {
            "participated_on": None,
            "state": None,
            "role": "REVIEWER",
            "user": {
                "display_name": "DefaultReviewer1",
                "uuid": "{DefaultReviewer1Uuid}",
                "links": {
                    "self": {"href": "bitbucket/cloud/2.0/users/%7BDefaultReviewer1Uuid%7D"},
                    "html": {"href": "https://bitbucket.org/%7BDefaultReviewer1Uuid%7D/"},
                    "avatar": {
                        "href": "https://secure.gravatar.com/avatar/31a164e6406a8deeb56e94c778030a9f?d=https%3A%2F%2Favatar-management--avatars.us-west-2.prod.public.atl-paas.net%2Finitials%2FS-3.png"
                    },
                },
                "nickname": "DefaultReviewer1Nickname",
                "type": "user",
                "account_id": "DefaultReviewer1AccountId",
            },
            "type": "participant",
            "approved": False
        },
        {
            "participated_on": "2020-12-20T12:11:57.822940+00:00",
            "state": None,
            "role": "PARTICIPANT",
            "user": {
                "display_name": "DefaultReviewer2",
                "uuid": "{DefaultReviewer2Uuid}",
                "links": {
                    "self": {"href": "bitbucket/cloud/2.0/users/%7BDefaultReviewer2Uuid%7D"},
                    "html": {"href": "https://bitbucket.org/%7BDefaultReviewer2Uuid%7D/"},
                    "avatar": {
                        "href": "https://secure.gravatar.com/avatar/31a164e6406a8deeb56e94c77803123?d=https%3A%2F%2Favatar-management--avatars.us-west-2.prod.public.atl-paas.net%2Finitials%2FS-3.png"
                    },
                },
                "nickname": "DefaultReviewer2Nickname",
                "type": "user",
                "account_id": "DefaultReviewer2AccountId",
            },
            "type": "participant",
            "approved": False
        }
    ],m
```

Reply:
What about using following construct instead of the additional argument or to extend the `get_link` to check the existence of the key first and return None if it doesn't exists. 

```python
        if url is None:
            try:
                url = self.get_link("self")
            except KeyError:
                pass
```

Reply:
What if it is needed but somehow not received. Than it would not fail and return None . This could lead to a problem elsewhere. In my opinion a value should only tried to be determined if it was expected.

Reply:
I'm not sure if I understand your point correct. You mean that BB returns the correct response type but without the link to self? I think that's very unlikely.

Reply:
Yes. Maybe for technical reasons, bugs, incidents or whatever. This would not break.